.github/workflows/ci-app_cloud_e2e_test.yml

```diff
@@ -0,0 +1,179 @@
+name: cloud-testing
+
+# Used to run the e2e tests on lightning.ai
+
+on: # Trigger the workflow on push or pull request, but only for the master branch
+  push:
+    branches:
+      - "master"
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.head_ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
+
+jobs:
+  cloud-test:
+    name: Cloud Test
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        app_name:
+          - v0_app
+          - boring_app
+#          - quick_start  # TODO: fix this
+          - template_streamlit_ui
+          - template_react_ui
+          - template_jupyterlab
+          - idle_timeout
+          - collect_failures
+          - custom_work_dependencies
+          - drive
+          - payload
+    timeout-minutes: 35
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.8"
+
+      - name: Check for related changes
+        uses: dorny/paths-filter@v2
```

Comment:
nice, what happens if the change is not on the path, does it ends the job at this point?
cc: @carmocca

Reply:
@Borda actually after reading the docs. I updated the workflow as follows:
1- `file-filters.yml` contains multiple filters, we can define separately for PyTorch/ app/ or anything between.
2-`changes` job will once first for every PR and outputs what has been changed (the job to be moved to another file)
3- based on the output from the`changes` job, we can define on the job and step level what to run.

Reply:
yep, read the doc too, so it is not as magic as I hopped but still useful :)